Dockerfile

```diff
@@ -128,22 +128,13 @@ RUN apt update
 RUN apt install git -y
 
 # Azure database clients and sql-server uses pyodbc which requires unixODBC and 'ODBC Driver 17
-# for SQL Server' Microsoft's python 3.11 base image comes with msodbcsql18 but we are expecting to
+# for SQL Server' Microsoft's python 3.12 base image comes with msodbcsql18 but we are expecting to
 # use the msodbcsql17 driver so need to install specific versions of some libraries and allow Docker
 # to downgrade some pre-installed packages.
 # Updating libgnutls30 to resolve CVE-2024-28835 and CVE-2024-28834.
 RUN apt-get update \
     && apt-get install -y gnupg gnupg2 gnupg1 curl apt-transport-https libgnutls30 \
-    && curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
-    && curl https://packages.microsoft.com/config/debian/10/prod.list \
-    > /etc/apt/sources.list.d/mssql-release.list \
-    && apt-get update \
-    && ACCEPT_EULA=Y apt-get install -y --allow-downgrades \
-    msodbcsql17 \
-    odbcinst=2.3.11-1 \
-    odbcinst1debian2=2.3.11-1 \
-    unixodbc-dev=2.3.11-1 \
-    unixodbc=2.3.11-1
+    && apt-get install -y msodbcsql17 odbcinst odbcinst1debian2 unixodbc-dev unixodbc
```

Comment:
I checked and it installs `2.3.11-3` of odbcinst, odbcinst1debian2, unixodbc-dev, unixodbc in this case. Using `mcr.microsoft.com/azure-functions/python:4-python3.12` `2.3.11-1` option was not available. Should that be fine? @mrostan

Reply:
There's a nifty Dockerfile linter that checks for best practice adherence I use called `hadolint`. One thing I recommends is that we pin version for `apt-get install` commands: https://github.com/hadolint/hadolint/wiki/DL3008

This is also recommended by Docker in their [Best practices documentation](https://docs.docker.com/build/building/best-practices/):

> Version pinning forces the build to retrieve a particular version regardless of what’s in the cache. This technique can also reduce failures due to unanticipated changes in required packages.

I think `2.3.11-3` is ok, but I would keep pinning, and just update the version to match what gets installed here.

Reply:
Also I think we should keep the `ACCEPT_EULA=Y` part as it's recommended by Microsoft: https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver16&tabs=alpine18-install%2Cdebian17-install%2Cdebian8-install%2Credhat7-13-install%2Crhel7-offline